file.c

```diff
@@ -236,8 +236,84 @@ int kibosh_read(const char *path UNUSED, char *buf, size_t size, off_t offset,
     } else {
         if (file->type == KIBOSH_FILE_TYPE_NORMAL) {
             fault = kibosh_fs_check_read_fault(fs, file->path);
+
+            // inject read_corrupt fault
+            if (fault >= CORRUPT_ZERO) {
+                int i;
+                double fraction = 0.0;
+                char *suffix = NULL;
+                struct kibosh_fault_base **iter;
+                struct kibosh_fault_read_corrupt *corrupt_fault;
+                for (iter = fs->faults->list; *iter; iter++) {
+                    if (strcmp((*iter)->type, KIBOSH_FAULT_TYPE_READ_CORRUPT) == 0) {
+                        corrupt_fault = (struct kibosh_fault_read_corrupt*)(*iter);
+                        fraction = corrupt_fault->fraction;
+                        suffix = corrupt_fault->suffix;
+                    }
+                }
+
+                int pos = (int) ((1.0 - fraction) * ret);
+
+                switch(fault) {
+                    case CORRUPT_RAND:
+                        for (i=0; i < ret; i++) {
+                            if (random_fraction() <= fraction) {
+                                memset(buf+i, (int) round(random_fraction() * 255.0), 1);
+                            }
+                        }
+                        break;
+
+                    case CORRUPT_ZERO:
+                        for (i=0; i < ret; i++) {
+                            if (random_fraction() <= fraction) {
+                                memset(buf+i, 0, 1);
+                            }
+                        }
+                        break;
+
+                    case CORRUPT_RAND_SEQ: ;
+                        int num_bytes = ret - pos;
+                        char *random_buf = malloc(num_bytes);
+                        for (i=0; i < num_bytes; i++) {
+                            random_buf[i] = (int) round(random_fraction() * 255.0);
+                        }
+                        memcpy(buf+pos, random_buf, num_bytes);
+                        free(random_buf);
+                        break;
+
+                    case CORRUPT_ZERO_SEQ:
+                        memset(buf+pos, 0, ret - pos);
+                        break;
+
+                    case CORRUPT_DROP:
+                        ret = pos;
+                        break;
+
+                    // CORRUPT_ZERO
+                    default:
```

Comment:
if this is just intended for CORRUPT_ZERO then specify that, and don't have a default case.  the compiler will warn if a case is not covered for an enum type (and you should be using the enum type, as described earlier)

Reply:
Removed the default option.